### DISCUSSION TITLE
Example on optimization with PDE constraints

### DISCUSSION THREAD

**Discussion by Song921012** [infiniteopt/InfiniteOpt.jl#272] [General]
Hello, I read your very nice paper

`Pulsipher J L, Zhang W, Hongisto T J, et al. A unifying modeling abstraction for infinite-dimensional optimization[J]. Computers & Chemical Engineering, 2022, 156: 107567.`

It said InfiniteOpt.jl can solve  PDE optimization problems. I read the documents, but I didn't find examples. How to use InfiniteOpt.jl to solve optimization problems with PDE constraints?  Discretize PDE to ODE (e.g., use DiffEqOperators), then use InfiniteOpt.jl?

Thank you for your help.

**Comment by pulsipher** [comment]
Welcome @Song921012 to the forum! I am glad you enjoyed the paper.

InfiniteOpt.jl can define PDE constraints directly using `@deriv` function which can be used to define a variety of partial derivatives. Note that currently, only square domains are natively supported (though other domains could be readily added via a small extension).

If you have a PDE in particular that you are looking to solve, feel free to share it here and I can give you more specific help.

**Reply by pulsipher** [reply]
On the development plan is the idea to connect to Gridap.jl to support more complex geometries and a couple other enhancements to improve our support of PDEs.

**Reply by Song921012** [reply]
Appreciate your help. Your answers are totally what I need.

**Reply by pulsipher** [reply]
Glad to be of help, please feel free to come back with any further questions you have.
